src/type-logic.ts

```diff
@@ -207,11 +207,15 @@ export function meet(a: Type, b: Type): Type {
     // union is join. meet distributes over join.
     return a.of.map(t => meet(t, b)).reduce(join);
   }
-  if (
-    (a.kind === 'list' && b.kind === 'list') ||
-    (a.kind === 'normal completion' && b.kind === 'normal completion')
-  ) {
-    return { kind: a.kind, of: meet(a.of, b.of) };
+  if (a.kind === 'list' && b.kind === 'list') {
+    return { kind: 'list', of: meet(a.of, b.of) };
+  }
+  if (a.kind === 'normal completion' && b.kind === 'normal completion') {
+    const inner = meet(a.of, b.of);
+    if (inner.kind === 'never') {
+      return { kind: 'never' };
+    }
```

Comment:
Doesn't this same thing apply to `list` above? Why handle these differently?

Reply:
The empty list is an inhabitant of `List<never>`, but you can't have an empty normal completion, so there are no inhabitants of `NormalCompletion<never>`.

Before we split "abrupt completion" and "normal completion" into distinct types (in the previous PR) there was just `Completion<never>`, which held abrupt completions, but now we can do better. Probably this change should have been in the previous PR but whatever.

Reply:
Ah I forgot about the empty list! 😖